runelite-client/src/main/java/net/runelite/client/plugins/fps/FpsPlugin.java

```diff
@@ -102,4 +122,66 @@ protected void shutDown() throws Exception
 	{
 		drawManager.unregisterEveryFrameListener(drawListener);
 	}
+
+	@Schedule(
+		asynchronous = true,
```

Comment:
This will block everything that is using the ScheduledExecutorService every 5 seconds, not sure if it should be done like this, maybe yes.

Reply:
Do you have any suggestions to change it to? I was initially using 5 seconds just to make it so I didn't have to wait a long time for it to update while testing. I'm opening to changing it to a longer period if that's something you would like

Reply:
Given that you're doing blocking network operations on it, I'd personally go with a separate single thread `ScheduledExecutorService` while draw ping is set. You'd need to do a slight bit of book-keeping for the starting and closing of the service, but it seems better over potentially blocking a lot of other operations using the main executor.

Reply:
Would you mind taking a look at what I've added?